src/main/kotlin/com/zepben/ewb/database/sqlite/cim/CimDatabaseReader.kt

```diff
@@ -59,14 +59,14 @@ abstract class CimDatabaseReader<TTables : CimDatabaseTables, TService : BaseSer
     /**
      * Read the database.
      */
-    fun read(service: TService): Boolean =
+    fun read(service: TService, afterServiceReadFunction: Boolean? = true): Boolean =
         try {
             require(!hasBeenUsed) { "You can only use the database reader once." }
             hasBeenUsed = true
 
             beforeRead()
                 && readService(service)
-                && afterServiceRead(service)
+                && if (afterServiceReadFunction ?: true) afterServiceRead(service) else true
```

Comment:
todo: No tests to prove this works, nor change log entry to indicate the new functionality.